fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java

```diff
@@ -607,21 +608,25 @@ public static int computeMinGEPower2(int num) {
     }
 
     public static boolean canEliminateNull(Set<ColumnRefOperator> nullOutputColumnOps, ScalarOperator expression) {
-        Map<ColumnRefOperator, ScalarOperator> m = nullOutputColumnOps.stream()
-                .map(op -> new ColumnRefOperator(op.getId(), op.getType(), op.getName(), true))
-                .collect(Collectors.toMap(identity(), col -> ConstantOperator.createNull(col.getType())));
-
-        for (ScalarOperator e : Utils.extractConjuncts(expression)) {
-            ScalarOperator nullEval = new ReplaceColumnRefRewriter(m).rewrite(e);
-
-            ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
-            // Call the ScalarOperatorRewriter function to perform constant folding
-            nullEval = scalarRewriter.rewrite(nullEval, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
-            if (nullEval.isConstantRef() && ((ConstantOperator) nullEval).isNull()) {
-                return true;
-            } else if (nullEval.equals(ConstantOperator.createBoolean(false))) {
-                return true;
+        try {
+            Map<ColumnRefOperator, ScalarOperator> m = nullOutputColumnOps.stream()
+                    .map(op -> new ColumnRefOperator(op.getId(), op.getType(), op.getName(), true))
+                    .collect(Collectors.toMap(identity(), col -> ConstantOperator.createNull(col.getType())));
+            for (ScalarOperator e : Utils.extractConjuncts(expression)) {
+                ScalarOperator nullEval = new ReplaceColumnRefRewriter(m).rewrite(e);
+
+                ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
+                // Call the ScalarOperatorRewriter function to perform constant folding
+                nullEval = scalarRewriter.rewrite(nullEval, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
+                if (nullEval.isConstantRef() && ((ConstantOperator) nullEval).isNull()) {
+                    return true;
+                } else if (nullEval.equals(ConstantOperator.createBoolean(false))) {
+                    return true;
+                }
             }
+        } catch (Exception e) {
+            LOG.warn("Failed to eliminate null: ", DebugUtil.getStackTrace(e));
```

Comment:
Had met exception thrown here?

Reply:
Yes,`test_transparent_mv_iceberg_part2` will always reproduce this it's because of 

`ScalarOperator operator = SqlToScalarOperatorTranslator.translate(expr);`  has met `DictMappingOperator` which will cause null result.

maybe we can fix this later.
```
2024-06-04 15:04:20.918+08:00 WARN (starrocks-mysql-nio-pool-0|250) [Utils.canEliminateNull():635] Failed to eliminate null: java.lang.NullPointerException: Cannot invoke "com.starrocks.sql.optimizer.operator.scalar.ScalarOperator.clone()" because "p" is null
        at com.starrocks.sql.optimizer.operator.scalar.PredicateOperator.lambda$clone$0(PredicateOperator.java:115)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at com.starrocks.sql.optimizer.operator.scalar.PredicateOperator.clone(PredicateOperator.java:115)
        at com.starrocks.sql.optimizer.rewrite.ReplaceColumnRefRewriter.rewrite(ReplaceColumnRefRewriter.java:47)
        at com.starrocks.sql.optimizer.Utils.canEliminateNull(Utils.java:622)
        at com.starrocks.planner.PlanNode.canEliminateNull(PlanNode.java:892)
        at com.starrocks.planner.OlapScanNode.lambda$canEliminateNull$21(OlapScanNode.java:1356)
        at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
        at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
        at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
        at com.starrocks.planner.OlapScanNode.canEliminateNull(OlapScanNode.java:1356)
        at com.starrocks.planner.PlanNode.couldBound(PlanNode.java:876)
        at com.starrocks.planner.PlanNode.pushDownRuntimeFilters(PlanNode.java:828)
        at com.starrocks.planner.SortNode.buildRuntimeFilters(SortNode.java:171)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.buildPartialTopNFragment(PlanFragmentBuilder.java:2486)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalTopN(PlanFragmentBuilder.java:2362)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalTopN(PlanFragmentBuilder.java:408)
        at com.starrocks.sql.optimizer.operator.physical.PhysicalTopNOperator.accept(PhysicalTopNOperator.java:144)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visit(PlanFragmentBuilder.java:456)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalTopN(PlanFragmentBuilder.java:2356)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalTopN(PlanFragmentBuilder.java:408)
        at com.starrocks.sql.optimizer.operator.physical.PhysicalTopNOperator.accept(PhysicalTopNOperator.java:144)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visit(PlanFragmentBuilder.java:456)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalDecode(PlanFragmentBuilder.java:711)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalDecode(PlanFragmentBuilder.java:408)
        at com.starrocks.sql.optimizer.operator.physical.PhysicalDecodeOperator.accept(PhysicalDecodeOperator.java:153)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visit(PlanFragmentBuilder.java:456)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.buildSetOperation(PlanFragmentBuilder.java:3033)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalUnion(PlanFragmentBuilder.java:3089)
        at com.starrocks.sql.plan.PlanFragmentBuilder$PhysicalPlanTranslator.visitPhysicalUnion(PlanFragmentBuilder.java:408)
        at com.starrocks.sql.optimizer.operator.physical.PhysicalUnionOperator.accept(PhysicalUnionOperator.java:69)
```